docs/how-to-guides/monitoring-with-prometheus-operator.md

```diff
@@ -0,0 +1,135 @@
+
+<!---
+TODO: Once we have a tutorial on how to deploy and configure Contour and cert-manager, point the following to that tutorial.
+-->
+
+  * [`metallb`](./ingress-with-contour-metallb.md) (only on Packet and bare-metal)
+  * [`contour`](../configuration-reference/components/contour.md)
+  * [`cert-manager`](../configuration-reference/components/cert-manager.md)
+
+
+## Deploy Prometheus Operator
+
+### Configure Prometheus Operator
+
+Create a file named `monitoring.lokocfg` with the following contents:
+
+```tf
+component "prometheus-operator" {}
+```
+
+For information about all the available configuration options for the `prometheus-operator` component, visit the component's [configuration reference](../configuration-reference/components/prometheus-operator.md).
+
+### Install Prometheus Operator
+
+Execute the following command to deploy the `prometheus-operator` component:
+
+```bash
+lokoctl component apply prometheus-operator
+```
+
+Verify the pods in the `monitoring` namespace are in the `Running` state (this may take a few minutes):
+
+```bash
+kubectl -n monitoring get pods
+```
+
+## Accessing Prometheus, Alertmanager and Grafana
+
+### Accessing Prometheus
+
+Execute the following command to forward port `9090` locally to the Prometheus pod:
+
+```bash
+kubectl -n monitoring port-forward svc/prometheus-operator-prometheus 9090
+```
+
+Open the following URL: [http://localhost:9090](http://localhost:9090).
+
+### Accessing Alertmanager
+
+Execute the following command to forward port `9093` locally to the Alertmanager pod:
+
+```bash
+kubectl -n monitoring port-forward svc/prometheus-operator-alertmanager 9093
+```
+
+Open the following URL: [http://localhost:9093](http://localhost:9093).
+
+### Accessing Grafana
+
+#### Using port forward
+
+Execute the following command to forward port `8080` locally to the Grafana dashboard pod on port `80`:
+
+```bash
+kubectl -n monitoring port-forward svc/prometheus-operator-grafana 8080:80
+```
+
+Obtain the password for the `admin` Grafana user by running the following command:
+
+```bash
+kubectl -n monitoring get secret prometheus-operator-grafana -o jsonpath='{.data.admin-password}' | base64 -d && echo
+```
+
+Open the following URL: [http://localhost:8080](http://localhost:8080). Enter the username `admin` and password obtained from the previous step.
+
+#### Using Ingress
+
+To expose Grafana to the internet using Ingress, provide the host field. The configuration for Grafana in the `prometheus-operator` component should look like the following:
+
+```tf
+component "prometheus-operator" {
+  grafana {
+    ingress {
+      host = "grafana.<cluster name>.<DNS zone>"
+    }
+  }
+}
+```
+
+> **NOTE**: On Packet, create a DNS entry for `grafana.<cluster name>.<DNS zone>` and point it to the Packet EIP as mentioned [here](./ingress-with-contour-metallb.md).
```

Comment:
I don't see anything DNS related here, not sure we have documentation on this yet.

Does it make sense to suggest using External DNS? (we'll have to check it works).

Reply:
We surely have to work upon our DNS story, at least remove the differences in how we suggest on Packet and on AWS. For AWS things work out of the box.

Reply:
I just added the ingress block to a prometheus-operator configuration of a cluster with the external DNS component installed, applied, and I could access the grafana dashboard through ingress without any further configuration.

How about we change this to the following?

```suggestion
> **NOTE**: On Packet, you either need to create a DNS entry for `grafana.<cluster name>.<DNS zone>` and point it to the Packet external IP for the contour service (see the [Packet ingress guide for more details](./ingress-with-contour-metallb.md)) or use the [External DNS component](../configuration-reference/components/external-dns.md).
```

Reply:
done.

Reply:
done.